app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,43 @@
 class PostsController < ApplicationController
-  def index
+  def find_user
     @user = User.find(params[:user_id])
   end
 
+  def index
+    find_user
+  end
+
   def show
-    @post = User.find(params[:user_id]).posts.find(params[:id])
+    @post = find_user.posts.find(params[:id])
+  end
+
+  def new
+    find_user
+    post = find_user.posts.new
+    render :new, locals: { post: post }
+  end
+
+  def create
+    find_user
+    new_post = find_user.posts.new(post_params)
```

Comment:
- [ ] Kindly use the `current_user` method created in the application controller.

Reply:
Hi @oloomoses ,

Thanks for this feedback.

Please I understand what you are saying, but I would like to explain something a bit. Imagine I am a user on Facebook and I create a post, should I be the one to comment on the post or other users should be the ones to comment on my post. That is the reason why I am using a random user to create the post, while the `current_user` is the one to comment or like the post. 

It would be nice if you considered this.

Thank you